code/game/objects/items/mail.dm

```diff
@@ -0,0 +1,313 @@
+	/// List of all stamp overlays on the letter.
+	var/list/stamps = list()
+	/// Maximum number of stamps on the letter.
+	var/stamp_max = 1
+	/// Physical offset of stamps on the object. X direction.
+	var/stamp_offset_x = 0
+	/// Physical offset of stamps on the object. Y direction.
+	var/stamp_offset_y = 2
+
+	///mail will have the color of the department the recipient is in.
+	var/static/list/department_colors
+
+/obj/item/mail/envelope
+	name = "envelope"
+	icon_state = "mail_large"
+	goodie_count = 2
+	stamp_max = 2
+	stamp_offset_y = 5
+
+/obj/item/mail/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_MOVABLE_DISPOSING, .proc/disposal_handling)
+	//AddElement(/datum/element/item_scaling, 0.75, 1)
+	if(isnull(department_colors))
+		department_colors = list(
+			ACCOUNT_CIV = COLOR_WHITE,
+			ACCOUNT_ENG = COLOR_PALE_ORANGE,
+			ACCOUNT_SCI = COLOR_PALE_PURPLE_GRAY,
+			ACCOUNT_MED = COLOR_PALE_BLUE_GRAY,
+			ACCOUNT_SRV = COLOR_PALE_GREEN_GRAY,
+			ACCOUNT_CAR = COLOR_BEIGE,
+			ACCOUNT_SEC = COLOR_PALE_RED_GRAY,
+		)
+
+	// Icons
+	// Add some random stamps.
+	if(stamped == TRUE)
+		var/stamp_count = rand(1, stamp_max)
+		for(var/i in 1 to stamp_count)
+			stamps += list("stamp_[rand(2, 6)]")
+	update_icon()
+
+/obj/item/mail/update_icon()
+	. = ..()
+	var/bonus_stamp_offset = 0
+	for(var/stamp in stamps)
+		var/image/stamp_image = image(
+			icon = icon,
+			icon_state = stamp,
+			pixel_x = stamp_offset_x,
+			pixel_y = stamp_offset_y + bonus_stamp_offset
+		)
+		stamp_image.appearance_flags |= RESET_COLOR
+		add_overlay(stamp_image)
+		bonus_stamp_offset -= 5
+
+	if(postmarked == TRUE)
+		var/image/postmark_image = image(
+			icon = icon,
+			icon_state = "postmark",
+			pixel_x = stamp_offset_x + rand(-3, 1),
+			pixel_y = stamp_offset_y + rand(bonus_stamp_offset + 3, 1)
+		)
+		postmark_image.appearance_flags |= RESET_COLOR
+		add_overlay(postmark_image)
+
+/obj/item/mail/attackby(obj/item/W, mob/user, params)
+	// Destination tagging
+	if(istype(W, /obj/item/destTagger))
+		var/obj/item/destTagger/destination_tag = W
+
+		if(sort_tag != destination_tag.currTag)
+			var/tag = uppertext(GLOB.TAGGERLOCATIONS[destination_tag.currTag])
+			to_chat(user, span_notice("*[tag]*"))
+			sort_tag = destination_tag.currTag
+			playsound(loc, 'sound/machines/twobeep_high.ogg', 100, TRUE)
+
+/obj/item/mail/attack_self(mob/user)
+	if(recipient_ref)
+		var/datum/mind/recipient = recipient_ref.resolve()
+		// If the recipient's mind has gone, then anyone can open their mail
+		// whether a mind can actually be qdel'd is an exercise for the reader
+		if(recipient && recipient != user?.mind)
+			to_chat(user, span_notice("You can't open somebody else's mail! That's <em>illegal</em>!"))
+			return
+
+	to_chat(user, span_notice("You start to unwrap the package..."))
+	if(!do_after(user, 1.5 SECONDS, target = user))
+		return
+	user.temporarilyRemoveItemFromInventory(src, TRUE)
+	if(contents.len)
+		user.put_in_hands(contents[1])
```

Comment:
Do you intend to just delete any other items in the mail (envelopes can spawn 2)